libs/injection/mlpe/injection/injection.py

```diff
@@ -1,81 +1,94 @@
+
+        # calculate waveform with lalsimulation
+        hplus, hcross = BurstSineGaussian(*sample, delta_t=delta_t)
+        hplus = hplus.data.data
+        hcross = hcross.data.data
+
+        # pad with zeros to desired waveform duration
+        waveform_size = int(sample_rate * duration)
+
+        total_pad = waveform_size - len(hplus)
+        pad_left = total_pad // 2
+        pad_right = total_pad - pad_left
+
+        hplus = np.pad(hplus, (pad_left, pad_right), mode="constant")
+        hcross = np.pad(hcross, (pad_left, pad_right), mode="constant")
+
+        cross.append(hcross)
+        plus.append(hplus)
+
+    cross = np.stack(cross)
+    plus = np.stack(plus)
+    return cross, plus
+
+
+def sine_gaussian_frequency(
+    frequency_array: np.ndarray,
+    hrss: float,
+    quality: float,
+    frequency: float,
+    phase: float,
+    eccentricity: float,
+    **kwargs,
+):
+
     """
+    Frequency domain sine gaussian built on top of
+    lalinference BurstSineGaussianF.
 
-    sample_params = [
-        dict(zip(sample_params, col)) for col in zip(*sample_params.values())
-    ]
+    Meant as a wrapper to use with bilby PE.
 
-    n_samples = len(sample_params)
+    Args:
+        frequency_array: frequencies at which to evaluate model
+        hrss: amplitude
+        q: quality factor
+        frequency: central frequency of waveform
+        phase: phase of waveform
+        eccentricity: relative fraction of hplus / hcross
 
-    waveform_generator = WaveformGenerator(
-        duration=waveform_duration,
-        sampling_frequency=sample_rate,
-        frequency_domain_source_model=waveform,
-        waveform_arguments=waveform_arguments,
-        parameter_conversion=parameter_conversion,
-    )
+    """
 
-    waveform_size = int(sample_rate * waveform_duration)
+    # number of frequencies
+    n_freqs = len(frequency_array)
 
-    num_pols = 2
-    signals = np.zeros((n_samples, num_pols, waveform_size))
+    # infer df from frequency array
+    df = frequency_array[1] - frequency_array[0]
 
-    for i, p in enumerate(sample_params):
-        polarizations = waveform_generator.time_domain_strain(p)
-        polarization_names = sorted(polarizations.keys())
-        polarizations = np.stack(
-            [polarizations[p] for p in polarization_names]
-        )
+    # infer dt from nyquist
+    dt = 1 / (2 * frequency_array[-1])
+
+    # calculate hplus and hcross from lalinference
+    hplus, hcross = BurstSineGaussianF(
+        quality, frequency, hrss, eccentricity, phase, df, dt
+    )
+    cross = np.zeros(n_freqs, dtype=complex)
+    plus = np.zeros(n_freqs, dtype=complex)
 
-        # center so that coalescence time is middle sample
-        dt = waveform_duration / 2
-        polarizations = np.roll(polarizations, int(dt * sample_rate), axis=-1)
-        signals[i] = polarizations
+    cross[: len(hcross.data.data)] = hcross.data.data
+    plus[: len(hplus.data.data)] = hplus.data.data
```

Comment:
why is this assigned relative to the start index of  `plus` or `cross`?

Reply:
Colm sent me this code at some point. I will look into the details of all of this.